crypto/mem_sec.c

```diff
@@ -433,8 +437,13 @@ static int sh_init(size_t size, int minsize)
     if (mprotect(sh.map_result + aligned, pgsize, PROT_NONE) < 0)
         ret = 2;
 
+#if defined(MLOCK_ONFAULT) && defined(SYS_mlock2)
+    if (syscall(SYS_mlock2, sh.arena, sh.arena_size, MLOCK_ONFAULT) < 0)
+        ret = 2;
+#else
```

Comment:
I think the 80% case will be to have compiled OpenSSL on the system to be used, or as part of a distribution package. However, it's always possible to have updated the OpenSSL package and not the kernel...